### ISSUE TITLE
[HOLD for payment 2024-05-15] HIGH: [Polish] [$500] Redesign thread ancestry

### ISSUE THREAD

**Issue by quinthar** [Expensify/App#36752]
Problem:
====
Infinite threading means you can get buried deep in a hierarchy and lose track of where you are.  We show your ancestors to help with this, but user complaints suggest this isn't enough:

![Image](https://github.com/Expensify/App/assets/575186/a0ac4dbc-25f1-4c73-a2f3-d0c414cf33cc)

Solution:
====
Make it look more clear!  There was a log discussion on this ([Slack](https://expensify.slack.com/archives/C066HJM2CAZ/p1708059588802499?thread_ts=1707431575.703659&cid=C066HJM2CAZ)) ending with this design:

![Image](https://github.com/Expensify/App/assets/575186/8496ad6d-9f78-41d9-9e76-5e608846f51c)

Specifically:

* Add a blue "thread" link to each separator line
* Add a gray "replies" message to the final separator
* From [below](https://github.com/Expensify/App/issues/36752#issuecomment-1967659135) - we want the entire message (including the blue thread icon) to be clickable to navigate you to the thread.
* From [below](https://github.com/Expensify/App/issues/36752#issuecomment-1979025046) - clicking on an ancestor message should take you to the thread/room where the message _originally appeared_, not the thread for the message.

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c49e505032aeb5f6</li>
        <li>Upwork Job ID: 1758664864087191552</li>
        <li>Last Price Increase: 2024-03-16</li>
<li>Automatic offers: </li>
<ul>
<li>aimane-chnaif | Reviewer | 0</li>
<li>rayane-djouah | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by dannymcclain** [comment]
[Figma file here.](https://www.figma.com/file/cUds2Gpyb0ve67fk1Zda9j/Threads?type=design&node-id=1771-115857&mode=design&t=NOdq7UiOboWCAkJi-11) cc'ing @Expensify/design on this for visibility.

And here's one more mockup to show more context around how ancestry works:
![image](https://github.com/Expensify/App/assets/7516624/956f20d0-30f9-4ac2-81b1-1ca1e264eab8)

This look right to you `@designteam`?

**Comment by shawnborton** [comment]
That looks right to me, thanks for laying it out like that!

**Comment by mallenexpensify** [comment]
@aimane-chnaif can you please review the proposals above? 
Added `NewFeature` to assign a BZ to help move this forward. 👋 @dylanexpensify .

**Comment by dylanexpensify** [comment]
Nice nice, thanks Matt!

**Comment by mallenexpensify** [comment]
> Ah this is weekly so missed from my radar. 2 proposals to review. will update on Monday

New Features default to `Weekly`, I didn't notice that either.  I bumped to `Daily`.

**Comment by Victor-Nyagudi** [comment]
I apologize in advance for the lack of proper formatting on the code blocks. 

My computer is at 3%, and I'm nowhere near a charging source. I intend to format them correctly once I've recharged my device. 

I see the video is uploaded, but it's taking a while to load/play for me. Let me know if you can see it, or if I need to upload it again.

**Comment by dannymcclain** [comment]
Hey everyone, one thing that I don't think got noted in the issue description is that we want the entire message (including the blue thread icon) to be clickable to navigate you to the thread.

**Comment by mallenexpensify** [comment]
Added your comment to the OP too
<img width="838" alt="image" src="https://github.com/Expensify/App/assets/22508304/ebe0c510-c87c-4cbe-b7c7-1658989aaf23">

**Comment by dylanexpensify** [comment]
cc everyone for above!

**Comment by Victor-Nyagudi** [comment]
## Proposal 

[Updated](https://github.com/Expensify/App/issues/36752#issuecomment-1967402994)

**Comment by Victor-Nyagudi** [comment]
> Hey everyone, one thing that I don't think got noted in the issue description is that we want the entire message (including the blue thread icon) to be clickable to navigate you to the thread.

I'm curious about two things here, @dannymcclain.

1. The thread ancestor message itself is already quite a large clickable/tapable area, so I wonder if maybe the video in my initial proposal where I clicked on the thread icon to navigate to the ancestor chat gave you the wrong impression. 
2. Should we go through with the icon being clickable, will there be any tooltips on it? The three dot button in the header has a tooltip, so will that pattern also be followed here?

**Comment by dannymcclain** [comment]
Hopefully this crude mock helps explain it better. I'm just saying that both the pink-overlayed area AND the orange-overlayed area should be clickable, and should navigate you to the thread.

![CleanShot 2024-02-28 at 15 43 12@2x](https://github.com/Expensify/App/assets/7516624/7f34e7fb-68f7-45a1-bc6e-9ef2cc592270)

I don't think we need any tooltips on the icon because we have the word "thread" right next to it.

**Comment by dannymcclain** [comment]
The @Expensify/design team and I were talking about something similar earlier—and basically just trying to figure out the best way for this to work. Would love to get their thoughts on your [proposed behavior video](https://github.com/Expensify/App/issues/36752#issuecomment-1969985202). I _think_ I think it makes more sense?

Also, as part of this, shouldn't we be updating the "From..." subheading in the chat header so that it accurately reflects where the thread is coming from?

**Comment by dubielzyk-expensify** [comment]
Hmm. This is a tricky one. Cause in some way I'm actually expecting that clickign on the message underneath the thread link will take me to where that message originated from. So basically in the third screen you can't interact with the message, only tap it and then it goes to the second screen. I worry the thread link is too small of a tap target. 

I'm happy to lean on @dannymcclain 's intuition here

**Comment by dannymcclain** [comment]
Does something like this make more sense? Where the _current_ thread message is not tappable. Clicking on one of the "higher up" threads takes you to the room for that thread?

![image](https://github.com/Expensify/App/assets/7516624/36d8ecbe-b271-4028-a2bd-e3507f9f3182)

cc @Expensify/design ☝️ Should we take this to Slack to get some clarity?

**Comment by dannymcclain** [comment]
>Hmm. should we disable all actions on the message and hide context menus? It was not specified in OP.

I don't think we need to disable or hide anything for the messages.

**Comment by Victor-Nyagudi** [comment]
Why don't we do this. 

> Infinite threading means you can get buried deep in a hierarchy and lose track of where you are. We show your ancestors to help with this, but user complaints suggest this isn't enough:

The [original problem](https://github.com/Expensify/App/issues/36752#issue-2139724950) we're trying to tackle is users losing track of where they are in the hierarchy, and the solution is to make things "look" clearer. 

> Make it look more clear! There was a log discussion on this ([Slack](https://expensify.slack.com/archives/C066HJM2CAZ/p1708059588802499?thread_ts=1707431575.703659&cid=C066HJM2CAZ)) ending with this design:

I'm not sure if there have been user complaints about whether the current navigation to ancestor threads is a pain point or not, so what if we first tackled the visual part, which is  a "HIGH" priority, and then address if the thread linking is a problem or not?

> There was a log discussion on this ([Slack](https://expensify.slack.com/archives/C066HJM2CAZ/p1708059588802499?thread_ts=1707431575.703659&cid=C066HJM2CAZ))

It seems a long discussion was already held, so if time is of the essence, I think it would be a good idea to knock things out one at a time. 

But then again, if the behavior of clicking/tapping certain ancestor threads and which ones lead to what has always been an issue, I leave it to you to decide if further discussions are necessary or not, and if that should also be handled here.

**Comment by dubielzyk-expensify** [comment]
> Should we take this to Slack to get some clarity?

I think so. There's some details to be worked out first.

**Comment by dannymcclain** [comment]
Not overdue. Going to get a quick confirmation on expected behavior in Slack today.

**Comment by dannymcclain** [comment]
[Slacked.](https://expensify.slack.com/archives/C066HJM2CAZ/p1709567786793169)

**Comment by dannymcclain** [comment]
Ok coming off of the Slack thread I linked above, I think we are fully read to rock this thing forward. I will update the OP after posting this, but I'm going to post it here too.

Essentially what we've decided is that we want the ancestor message/thread you are currently in to be clickable, and it should take you to the room/thread where the message originally appeared.

So clicking on an ancestor message should take you to where the message was originally posted—not the thread of that message.

**Basic mocks:**
![image](https://github.com/Expensify/App/assets/7516624/3b61706a-715c-43a1-815a-20d1d5bc7cee)

**Clickable area:**
![image](https://github.com/Expensify/App/assets/7516624/cfa0e54e-076c-4f6f-a7df-3ff116006210)

**Desired interaction:**
![image](https://github.com/Expensify/App/assets/7516624/920aea7b-e9d6-44b1-98b6-b94427d466b4)

cc @Expensify/design because this has been a ride haha.

**Comment by Victor-Nyagudi** [comment]
## Proposal 

[Updated](https://github.com/Expensify/App/issues/36752#issuecomment-1967402994).

**Comment by dylanexpensify** [comment]
Nice, @aimane-chnaif to review!

**Comment by dylanexpensify** [comment]
@aimane-chnaif bump please

**Comment by dylanexpensify** [comment]
@aimane-chnaif bump please for update 🙇‍♂️

**Comment by dylanexpensify** [comment]
Any updates here @rayane-djouah @aimane-chnaif? Let's get it moving!

**Comment by quinthar** [comment]
@rayane-djouah can you give an ETA for completion?

**Comment by dylanexpensify** [comment]
Nice!

**Comment by chiragsalian** [comment]
Cool, i pushed the translations forward in our other channel and linked your slack post. Looks like we got it answered 🙂

**Comment by dylanexpensify** [comment]
@allroundexperts @rayane-djouah what's left here before we can merge?

**Comment by quinthar** [comment]
Who exactly are we waiting on to review?

**Comment by dannymcclain** [comment]
@Victor-Nyagudi That's a good catch. I don't think it's the end of the world, but would be nice to fix it to where "Thread" and "Replies" is not copy-able.

Personally I think that could be a follow up or polish issue though?

**Comment by dylanexpensify** [comment]
Nice catch @Victor-Nyagudi!

@rayane-djouah what's the status on follow up PR? Let's squash this before we finish this issue, eh?

**Comment by dannymcclain** [comment]
Not overdue Melvin. The follow up PR is on hold because of a bug on main.

**Comment by dylanexpensify** [comment]
bump @allroundexperts

**Comment by allroundexperts** [comment]
Hi @dylanexpensify. The PR is on hold currently.

**Comment by dannymcclain** [comment]
Still on hold as far as I know.

**Comment by dylanexpensify** [comment]
same

**Comment by dylanexpensify** [comment]
Same, holding on 39982

**Comment by dylanexpensify** [comment]
Similar

**Comment by dannymcclain** [comment]
Looks like [39982](https://github.com/Expensify/App/issues/39982) has been deployed.

**Comment by dylanexpensify** [comment]
NOICE!

**Comment by dylanexpensify** [comment]
PR not yet on staging, waiting

**Comment by dylanexpensify** [comment]
Payment summary:
- Contributor: @rayane-djouah $750
- Contributor+: @allroundexperts $750

Please apply/request!

**Comment by dylanexpensify** [comment]
updated to reflect

**Comment by dylanexpensify** [comment]
payment tomorrow!

**Comment by dylanexpensify** [comment]
@rayane-djouah we decided $750 given there was a follow on issue created and done that wasn't necessarily needed to solve the bug, but a polish. Does that sound right? cc @chiragsalian

**Comment by dannymcclain** [comment]
Not overdue, right @dylanexpensify?

**Comment by dylanexpensify** [comment]
Nope! Just finishing payments!

**Comment by dylanexpensify** [comment]
Done!

**Comment by JmillsExpensify** [comment]
$750 approved for @allroundexperts
